### PR TITLE
Backend changes - 8664

### PR DESCRIPTION
**Pull request by karthik-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/8664

*Description of changes:*

- created migration script for updating remarks field to array(postgres) and list(python)
- refactor task service layer accordingly
- updated tests

How it would be in DB
image

Before Upgrade
image
After upgrade
image
After downgrade
image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/migrations/versions/304705f06971_tasks_multiple_reasons.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    conn = op.get_bind()
+    task_res = conn.execute("SELECT * FROM tasks;")
+    task_list: List[Task] = task_res.fetchall()
+
+    for task in task_list:
+        if task.remarks is None:
+            continue
+        # A check if there are task remarks that is already wrapped in '{}', ideally, we would not hit this scenario
+        if task.remarks.startswith('{') and task.remarks.endswith('}'):
```
[the remark this answers is not in the document]
Comment (karthik-aot) [reply auth-api/migrations/versions/304705f06971_tasks_multiple_reasons.py:36]: yeah. I shall refactor the same way like downgrade, check if it an array :)

Comment (saravanpa-aot) [inline auth-api/migrations/versions/304705f06971_tasks_multiple_reasons.py:36]: you can remove the whole thing rite? its an upgrade script ;so we will never have this scenario
also select * from can check for tasks which remarks exists

File: auth-api/migrations/versions/304705f06971_tasks_multiple_reasons.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            task_id = task.id
+            remarks = task.remarks.split(',')
+            # Now updating the new remarks value back in the table
+            update_sql = text("UPDATE tasks SET remarks = :remarks where id = :task_id") \
+                .params(
+                remarks=remarks, task_id=task_id)
+            op.execute(update_sql)
+
+    # ### alter command auto generated by Alembic ###
+    op.alter_column('tasks', 'remarks',
```
Comment (saravanpa-aot) [inline auth-api/migrations/versions/304705f06971_tasks_multiple_reasons.py:48]: shudnt this happen before the update statement?

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -60,19 +60,19 @@ services:
...
+      mem_limit: 512m
+      expose:
+        - 4222
+        - 8222
+      labels:
+        - entity.services=nats
+      ports:
+        - 4222:4222
+        - 8222:8222
+      tty: true
```
Comment (saravanpa-aot) [inline docker/docker-compose.yml:76]: why is this change?

File: auth-api/migrations/versions/304705f06971_tasks_multiple_reasons.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # We get the list of tasks at first.
+    # we convert tasks.remarks from character type to array form ( check if there are arrays already, edge case )
+    # we convert the column remarks type from character to ARRAY
+
+    conn = op.get_bind()
```
[the remark this answers is not in the document]
Comment (karthik-aot) [reply auth-api/migrations/versions/304705f06971_tasks_multiple_reasons.py:28]: ARRAY type expects JSON value and would not accept string literal. 
image
Therefore, I transform the existing column value to JSON array type and then alter the column :)

File: auth-api/migrations/versions/304705f06971_tasks_multiple_reasons.py (hunk 1 above)
Comment (karthik-aot) [reply auth-api/migrations/versions/304705f06971_tasks_multiple_reasons.py:36]: yeah refactored the condition now

File: auth-api/migrations/versions/304705f06971_tasks_multiple_reasons.py (hunk 2 above)
Comment (karthik-aot) [reply auth-api/migrations/versions/304705f06971_tasks_multiple_reasons.py:48]: ARRAY type expects JSON value and would not accept string literal.

File: auth-api/migrations/versions/304705f06971_tasks_multiple_reasons.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (karthik-aot) [reply auth-api/migrations/versions/304705f06971_tasks_multiple_reasons.py:28]: Before Upgrade
image
After upgrade
image
After downgrade
image

**Review by sumesh-aot — APPROVED** [review]

**Review by saravanpa-aot — APPROVED** [review]

**Pull request merged**